src/context/cdf_context.rs

```diff
@@ -615,14 +619,20 @@ trait CDFContextLogOps: CDFContextLogSize {
   ) {
     let base = fc as *mut _ as *mut u8;
     let mut len = log.data.len();
+    debug_assert!(len % (Self::CDF_LEN_MAX + 1) == 0);
     // SAFETY: We use unchecked pointers here for performance.
     // Since we know the length, we can ensure not to go OOB.
     unsafe {
       let mut src = log.data.as_mut_ptr().add(len);
       while len > checkpoint {
+        debug_assert!(len >= Self::CDF_LEN_MAX + 1);
```

Comment:
Clippy is being pedantic:
```
error: unnecessary `>= y + 1` or `x - 1 >=`
```
```suggestion
        debug_assert!(len > Self::CDF_LEN_MAX);
```

Reply:
I think `>= Self::CDF_LEN_MAX + 1` represents the intent better, since we need to ensure that it's possible to remove `Self::CDF_LEN_MAX + 1` elements from the collection.

Reply:
I might go ahead an rework the log structs with const generics. Then these assertions can be simplified or elided.

Reply:
I prefer the way clippy suggests though.

Reply:
As above, please take a look at the latest code.